sdks/python/apache_beam/dataframe/frames_test.py

```diff
@@ -122,7 +123,12 @@ def _run_test(self, func, *args, distributed=True, nonparallel=False):
             generated twice, once outside of an allow_non_parallel_operations
             block (to verify NonParallelOperation is raised), and again inside
             of an allow_non_parallel_operations block to actually generate an
-            expression to verify."""
+            expression to verify.
+        check_proxy (bool): Whether or not to check that the proxy of the
+            generated expression matches the actual result, defaults to True.
```

Comment:
For my education, what are the implications of us currently having mismatch proxies? (I am not very familiar with how proxy are actually used, and when they are generated)